### DISCUSSION TITLE
Wix v4 - Does the "Uninstall" usually work like this, where it shows the Resume Dialog with "Install" Button

### DISCUSSION THREAD

**Discussion by garevaul** [wixtoolset/issues#7351] [Questions]
Hi 

Does the "Uninstall" usually work like this, where it shows the **Resume Dialog** with the "Install" Button?

Is this Correct, Logically I think if we are coming from the "Uninstall" button, then it should display "Uninstall" instead of "Install" ?

Regards

![image](https://user-images.githubusercontent.com/15125247/229500388-4fe4b656-86a5-4efc-8833-cd414673e771.png)

![image](https://user-images.githubusercontent.com/15125247/229500442-fbf8a33f-96fc-48ea-b2e3-e4979cc3acea.png)

**Comment by MiekVenter** [comment]
As stated in the FAQ in https://wixtoolset.org/docs/fourthree/faqs/  in MsiPackage  with the namespace 
xmlns:bal="http://wixtoolset.org/schemas/v4/wxs/bal"

use the following.
<MsiPackage bal:DisplayInternalUICondition="WixBundleExecutePackageAction = 2" ....

then it words for me exactly as in v3.x before upgrading to wix4, the Setup dialog does not show during uninstall.

Tested it with supplied UI <ui:WixUI Id="WixUI_FeatureTree"/>

**Comment by MiekVenter** [comment]
No, my previous statement is not correct. the Setup will not display at all with the "WixBundleExecutePackageAction = 2" option.  There is two outcomes, you either have a setup dialog everywhere, or nowhere. The "install only" setup dialog of V3.x I cannot get working on wix4.  We are reverting to wix3 until there is better documentation or someone figure it out, or  a fix.
